scipy/linalg/tests/test_lapack.py

```diff
@@ -2656,14 +2656,9 @@ def test_ptsvx_error_raise_errors(dtype, realtype, fact, df_de_lambda):
     df, ef, info = df_de_lambda(d, e)
 
     # test with malformatted array sizes
-    with assert_raises(ValueError):
-        ptsvx(d[:-1], e, b, fact=fact, df=df, ef=ef)
-    with assert_raises(ValueError):
-        ptsvx(d, e[:-1], b, fact=fact, df=df, ef=ef)
-    df, ef, x, rcond, ferr, berr, info = ptsvx(d, e, b[:-1], fact=fact,
-                                               df=df, ef=ef)
-    # this info returns -9
-    assert info < 0
+    assert_raises(ValueError, ptsvx, d[:-1], e, b, fact=fact, df=df, ef=ef)
+    assert_raises(ValueError, ptsvx, d, e[:-1], b, fact=fact, df=df, ef=ef)
+    assert_raises(Exception, ptsvx, d, e, b[:-1], fact=fact, df=df, ef=ef)
```

Comment:
Adding a check it won't return ValueError but with:

```python
>       ptsvx(d, e, b[:-1], fact=fact, df=df, ef=ef)
E       _flapack.error: (shape(b, 0) >= n) failed for 3rd argument b
```
is there a way to return ValueError instead?

Reply:
No f2py raises base `Exception` class exceptions. You can catch those.

Reply:
Search for `assert_raises(Exception` phrase to see examples of it in the same `test_lapack.py` file.

Reply:
Thanks. That's what we are doing now. And all the tests pass too 👍